tests/components/nightscout/test_config_flow.py

```diff
@@ -0,0 +1,108 @@
+"""Test the Nightscout config flow."""
+from aiohttp import ClientConnectionError
+
+from homeassistant import config_entries, data_entry_flow, setup
+from homeassistant.components.nightscout.const import DOMAIN
+from homeassistant.components.nightscout.utils import hash_from_url
+from homeassistant.const import CONF_URL
+
+from tests.async_mock import patch
+from tests.common import MockConfigEntry
+from tests.components.nightscout import GLUCOSE_READINGS, SERVER_STATUS
+
+CONFIG = {CONF_URL: "https://some.url:1234"}
+
+
+async def test_form(hass):
+    """Test we get the user initiated form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["errors"] == {}
+
+    with _patch_glucose_readings(), _patch_server_status(), _patch_async_setup() as mock_setup, _patch_async_setup_entry() as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], CONFIG,
+        )
+
+        assert result2["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+        assert result2["title"] == SERVER_STATUS.name  # pylint: disable=maybe-no-member
+        assert result2["data"] == CONFIG
+        await hass.async_block_till_done()
+        assert len(mock_setup.mock_calls) == 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)